src/js/components/award/shared/activity/ContractGrantsActivityChart.jsx

```diff
@@ -97,23 +121,117 @@ const ContractGrantsActivityChart = ({
         return updatedTicks;
     };
     /**
-     * createxScale
+     * xTickDateAndLabel
+     * - format the x-axis labels
+     * @param {Number[]} - an array of dates in milliseconds
+     * @returns {Object[]} - an array of objects with date and label properties
+     */
+    const xTickDateAndLabel = (ticks) => ticks.map((date) => {
+        const newDate = new Date(date);
+        const year = convertDateToFY(moment(date));
+        const shortYear = (year).toString().slice(-2);
+        const shortMonth = newDate.toLocaleString('en-us', { month: 'short' }).toUpperCase();
+        const label = `${shortMonth} FY '${shortYear}`;
+        return { date: newDate, label };
+    });
+    /**
+     * createXScaleAndTicks
      * - creates the x scaling function and updates state
      * @returns {null}
      */
-    // const createXScaleAndTicks = useCallback(() => {
-    //     const scale = scaleLinear().domain(xDomain).range([0, visualizationWidth]).nice();
-    //     const ticks = scale.ticks(6);
-    //     // add last tick for spacing
-    //     const updatedTicksWithSpacing = addTicksForSpacing(ticks, true);
-    //     // create new scale since we have new data
-    //     const updatedScale = scaleLinear()
-    //         .domain([xDomain[0], updatedTicksWithSpacing[updatedTicksWithSpacing.length - 1]])
-    //         .range([0, visualizationWidth])
-    //         .nice();
-    //     setXTicks(updatedTicksWithSpacing);
-    //     setXScale(() => updatedScale);
-    // }, [xDomain, visualizationWidth]);
+    const createXScaleAndTicks = useCallback(() => {
+        if (!xDomain.length) return null;
+        /**
+         * By design, we want to leave space between the start of the chart
+         * and the end of the chart.
+         */
+        const spacing = visualizationWidth * 0.05;
+        const scale = scaleLinear()
+            .domain(xDomain)
+            .range([spacing, visualizationWidth - spacing - padding.left]);
+        let theTicks = scale.ticks(6);
+        // add first and last tic per design
+        theTicks.splice(0, 0, xDomain[0]); // first tick
+        theTicks.push(xDomain[1]); // last tick
```

Comment:
So do we have 8 ticks then?

Reply:
Not necessarily. The d3 ticks method will not guarantee the number of ticks it provides to you, but you can give it a default and it will try to appease. From their repo, "The specified count is only a hint; the scale may return more or fewer values depending on the domain." - https://github.com/d3/d3-scale#continuous_ticks